Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -248,77 +233,12 @@ std::string LeptonTrackingSFProducer::GetProducerId() const { return "LeptonTrac
-    } 
-    
-    if(settings.GetChannel() == "mm"){
-        histoname = settings.GetLeptonTrackingSFHistogramName();
-    }
-    else{
-        LOG(ERROR) << "LeptonTrackingSFProducer not implemented for this channel";
-    }
-
-    if (m_reversed_axes)
-    {
-        m_etabins = &m_ybins;
-        m_ptbins = &m_xbins;
-    }
-    else
-    {
-        m_etabins = &m_xbins;
-        m_ptbins = &m_ybins;
-    }
-    // Get file
-    LOG(INFO) << "Loading lepton scale factors for Tracking: File " << m_sffile
-              << ", Histogram " << histoname;
-    TFile file(m_sffile.c_str(), "READONLY");
-    TGraphAsymmErrors *sfhisto = (TGraphAsymmErrors*)file.Get(histoname.c_str());
-
-    // Get the pT and eta bin borders
-    m_xbins.emplace_back(sfhisto->GetX()[0] - sfhisto->GetErrorXlow(0));
-    for (int ix = 0; ix < sfhisto->GetN(); ++ix){
-        m_xbins.emplace_back(sfhisto->GetX()[ix] + sfhisto->GetErrorXhigh(ix));
-    }
-    m_ybins.emplace_back(0);
-    m_ybins.emplace_back(10000);
-    
-    // Fill the m_sf array with the values from the root histos
-    for (int ix = 1; ix <= sfhisto->GetN(); ix++) {
-        m_sf[ix - 1][0] = static_cast<float>(sfhisto->GetY()[ix-1]);
-        m_er[ix - 1][0] = static_cast<float>(sfhisto->GetErrorY(ix-1));
-    }
-    delete sfhisto;
-    file.Close();*/
-}
+void LeptonTrackingSFProducer::Init(ZJetSettings const& settings) {}
 
 void LeptonTrackingSFProducer::Produce(ZJetEvent const& event,
-                                 ZJetProduct& product,
-                                 ZJetSettings const& settings) const
+                                       ZJetProduct& product,
+                                       ZJetSettings const& settings) const
 {
-    //some old SF files already include the inverse, make sure you use them the right way!
-    /*if(product.m_zValid){
-        product.m_weights["zl1TrackingSFWeight"] = 1/GetScaleFactor(0, 0, *product.m_zLeptons.first);
-        product.m_weights["zl2TrackingSFWeight"] = 1/GetScaleFactor(0, 0, *product.m_zLeptons.second);
-        product.m_weights["leptonTrackingSFWeight"] =
-                1/GetScaleFactor(0, 0, *product.m_zLeptons.first) * 1/GetScaleFactor(0, 0, *product.m_zLeptons.second);
-        if (settings.GetLeptonSFVariation() == true) {
-            product.m_weights["leptonTrackingSFWeightUp"] =
-                1/GetScaleFactor(0, -1, *product.m_zLeptons.first) * 1/GetScaleFactor(0, -1, *product.m_zLeptons.second);
-            product.m_weights["leptonTrackingSFWeightDown"] =
-                1/GetScaleFactor(0, 1, *product.m_zLeptons.first) * 1/GetScaleFactor(0, 1, *product.m_zLeptons.second);
-        }
-    }
-    else {
-        product.m_weights["leptonTrackingSFWeight"] = 0;
-        if (settings.GetLeptonSFVariation() == true) {
-            product.m_weights["leptonTrackingSFWeightUp"] = 0;
-            product.m_weights["leptonTrackingSFWeightDown"] = 0;
-        }
-    }*/
 }
```

Comment:
Same here. It seems there is something missing.

Reply:
Before it was empty as well, I removed the old commented code. Currently no tracking SFs are provided. I could add a log fatal.